GCS:Debug: Improve debug widget, connect usagestats

tracernz opened the pull request:
Configurable log level in the usage stats settings.

TODO: make sure DEFINES += QT_MESSAGELOGCONTEXT doesn't have a bad impact on release builds.

TODO in future: Make DebugGadget able to filter log messages by severity, source (class?). Then make stuff emit debug messages all the time rather than defined at compile time.

tracernz committed:
GCS:Debug: Improve debug widget, connect usagestats

mlyle commented:
jenkins, build artifacts please

tracernz commented:
Force-pushed without the random line from another changeset :(

jenkins, build artifacts please

mlyle commented:
```
/var/lib/jenkins/jobs/dronin/workspace@2/Nodes/master/ground/gcs/src/plugins/debuggadget/debuggadgetfactory.cpp: In function ‘void customMessageHandler(QtMsgType, const QMessageLogContext&, const QString&)’:
/var/lib/jenkins/jobs/dronin/workspace@2/Nodes/master/ground/gcs/src/plugins/debuggadget/debuggadgetfactory.cpp:58:116: error: ‘level’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
     DebugEngine::getInstance()->message(level, msg, QString(context.file), context.line, QString(context.function));
```

tracernz commented:
GCC is so dumb sometimes. 😢 CLang and MSVC both got it right. Interesting that it passed the normal build but not artifacts with GCC... :/

tracernz committed:
GCS:DebugGadget: Work around GCC fail

mlyle commented:
> . Interesting that it passed the normal build but not artifacts with GCC... :/

Optimizer

mlyle commented:
jenkins, build artifacts please

tracernz committed:
GCS:DebugGadget: Work around my fail

tracernz commented:
jenkins, build artifacts please

tracernz commented:
Should put timestamps on these so they can be sewn into the usage details if desired.

tracernz commented:
Something not quite right with the QMake files in UsageStats. Will check/correct this evening.

tracernz committed:
GCS: Fix messed up qmake files

Uploader erroneously included gcsplugin.pri in it's deps, which would break debug builds for any other plugin that correctly had gcsplugin.pri in it's .pro and included uploader.pri (because TARGET would already have the debug suffix the second time gcsplugin.pri was included, the build would require <plugin_name>_debug.pluginspec => fail). UsageStats omitted it completely and only managed to build properly because it included uploader.pri. Untangled all this mess so things will work properly.

tracernz committed:
GCS:UsageStats: Include timestamps in debug logs

tracernz commented:
jenkins, build artifacts please

This should be good now.

tracernz committed:
GCS:DebugGadget: Install message handler always

Not just when the widget exists in the layout. This is needed so that usagestats can still collect data even when the widget isn't in the users layouts.

tracernz commented:
Made another change. Now install the message handler always when the DebugGadget plugin loads, not just when the widget is created. This allows us to collect debug logs in UsageStats even when the DebugGadget widget isn't in the users workspaces.
